### ISSUE TITLE
The project path contains non-ASCII characters, causing the run to error

### ISSUE THREAD

**Issue by yincolor** [oven-sh/bun#6947]
### What version of Bun is running?

1.0.9

### What platform is your computer?

Linux mint 21.2 x86_64

### What steps can reproduce the bug?

If the folder path of a project created using the "bun init" command contains Chinese characters, Japanese katakana, or Korean letters, executing the "bun run index.ts" command will result in an error.

with the error message reading:

error: Cannot find module "" from "bun:main"

### Additional information

I guess  the project path contain non-ASCII characters, witch causing the error during the runtime.

**Comment by Jarred-Sumner** [comment]
can you paste exactly what characters?

we have tests for this, so i'm wondering what we're missing from the tests

https://github.com/oven-sh/bun/blob/c3a7b574c876bb84a926e8be4c65e1f0a69e85e5/test/js/bun/resolve/non-english-import.test.js#L33-L55

**Comment by yincolor** [comment]
> can you paste exactly what characters?
> 
> we have tests for this, so i'm wondering what we're missing from the tests
> 
> https://github.com/oven-sh/bun/blob/c3a7b574c876bb84a926e8be4c65e1f0a69e85e5/test/js/bun/resolve/non-english-import.test.js#L33-L55

[bun_init_run_in_terminal.txt](https://github.com/oven-sh/bun/files/13277707/bun_init_run_in_terminal.txt)
This is the result I got by repeating the previous operation in the terminal.

**Comment by Hanaasagi** [comment]
I extracted all commands from the txt, but I couldn't reproduce this issue locally.

```Bash
mkdir .dev
cd .dev
mkdir test1_en
mkdir test2_你好
mkdir test3_こんにちは
mkdir test4_안녕하세요
cd test1_en/
mkdir test1_project
cd test1_project/
bun init
bun run index.ts
cd ../../test2_你好/
mkdir test2_project
cd test2_project/
bun init
bun run index.ts
cd ../../test3_こんにちは/
mkdir test3_project
cd test3_project/
bun init
bun run index.ts
cd ../../test4_안녕하세요/
mkdir test4_project
cd test4_project/
bun init
bun run index.ts
```

**Comment by yincolor** [comment]
yes. This issue has been resolved, when I updated to version 1.0.17.
